lint/src/main/java/com/firebaseui/lint/NonGlobalIdDetector.kt

```diff
@@ -0,0 +1,52 @@
+package com.firebaseui.lint
+
+import com.android.tools.lint.detector.api.Category
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.LayoutDetector
+import com.android.tools.lint.detector.api.Scope
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.XmlContext
+import org.w3c.dom.Element
+
+/**
+ * Lint detector to find layout attributes that reference "@id" where they should reference
+ * "@+id" instead.
+ */
+class NonGlobalIdDetector : LayoutDetector() {
+    override fun getApplicableElements() = Detector.XmlScanner.ALL
```

Comment:
Yeah I knew there had to be a better way than using `visitDocument`.  The JavaDoc for these detectors are ... not great.

Reply:
Can't you just walk over to Tor's desk and glare at him for a bit? 😆 Dunno if you saw his awesome talk, but if not here it is: https://youtu.be/p8yX5-lPS6o.

Reply:
Yeah the documentation is rough. There's the [google group](https://groups.google.com/forum/#!forum/lint-dev). And all of the other checks are open source so I started reading some of them and got to know the api through that.